### PR TITLE
Add more sources to tasks.proj build semaphore

### PR DESCRIPTION
**Pull request by akoeplinger**
With the Apple/AndroidAppBuilders we now have additional sources besides *.cs so we need to add them to the semaphore,
otherwise changes in these files wouldn't cause a rebuild.

See also https://github.com/dotnet/runtime/issues/37783

### CODE REVIEW

File: src/tasks/tasks.proj [hunk 1]
```diff
@@ -49,7 +49,7 @@
     <!-- Include both the project file and its sources as an input. -->
     <ItemGroup>
       <TasksSrc Include="%(ProjectReferenceWithConfiguration.Identity)" />
-      <TasksSrc Include="%(ProjectReferenceWithConfiguration.RelativeDir)%(ProjectReferenceWithConfiguration.RecursiveDir)**\*.cs" />
+      <TasksSrc Include="%(ProjectReferenceWithConfiguration.RelativeDir)%(ProjectReferenceWithConfiguration.RecursiveDir)**\*" />
```
Comment (ViktorHofer) [inline src/tasks/tasks.proj:52]: That probably includes the .csproj file twice because of the line above.

Comment (akoeplinger) [reply src/tasks/tasks.proj:52]: yeah, does it matter though? I guess we could simplify the whole thing to `$(MSBuildThisFileDirectory)**\*` so we don't even need to rely on the `ProjectReferenceWithConfiguration `

Comment (ViktorHofer) [reply src/tasks/tasks.proj:52]: That would include everything which we don't want as some P2Ps are condition per TargetOS. I would just remove the line above.

Comment (akoeplinger) [reply src/tasks/tasks.proj:51]: Ok.

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**